### DISCUSSION TITLE
Indexing newly inserted rows

### DISCUSSION THREAD

**Discussion by LiuDaveLiu** [milvus-io/milvus#30774] [Q&A and General discussion] [answered]
I am inserting new rows one by one, and they do not seem to be index upon insertion. Any suggestion to have them indexed?

**Comment by yhmo** [comment, marked as answer]
Milvus manages data as segments. There are two kinds of segments: growing segment and sealed segment.
Newly inserted data is accumulated in growing segment. Once a growing segment's size reaches a threshold(by default 110+MB), the growing segment is converted to a sealed segment. Sealed segment's data is const, cannot be changed, but sealed segments can be merged to large segments.

Before version v2.3.4, only sealed segments can be indexed. 

When search request is received, the search engine travels segments one by one. For sealed segments that were indexed, the search engine uses index to do fast search. For sealed segments that are not indexed, the search engine performs brute-force search. For growing segments, the search engine performs brute-force search, too.
Each segment returns a topk results, and the search engine combine the N results into a final topk result.

From version v2.3.4, we provide a temp index for growing segment. The temp index is IVF index, the index takes effect once a growing segment's row count exceeds a threshold(by default, a few thousands of rows).
